src/asm/parser.y

```diff
@@ -421,27 +424,45 @@ endc		: T_POP_ENDC T_NEWLINE {
 		}
 ;
 
-scoped_id	: T_ID | T_LOCAL_ID ;
+scoped_id	: T_ID
+			| T_LOCAL_ID
+			| T_OP_EVAL T_LPAREN string T_RPAREN {
+				struct Symbol const *sym = sym_FindScopedSymbol($3);
+
+				if (!sym)
+					fatalerror("EVAL: Symbol does not exist '%s'\n", $3);
+
+				if (snprintf($$, MAXSYMLEN + 1, "%s", sym->name) > MAXSYMLEN)
+					fatalerror("EVAL: Symbol is too long '%s'\n", sym->name);
+			}
+			;
+
+sym_value	: T_LABEL
```

Comment:
Since this production holds the symbol name, I think calling it "value" is misleading; since `T_LABEL` is always used in definitions, maybe `sym_def_name`?

Reply:
Sure.